### PR TITLE
Configurable Tcp keepalive

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by lemonmojo** [comment]
Hi @akallabeth! Thx for the quick response and the PR.
There's an error though as the value for `TCP_KEEPIDLE` shouldn't be based on `FreeRDP_TcpKeepAliveInterval`. I think we'll need an additional setting for this.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by lemonmojo** [comment]
@akallabeth Looks good now, thx!

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/core/settings.c [hunk 1]
```diff
@@ -559,6 +559,12 @@ rdpSettings* freerdp_settings_new(DWORD flags)
 	if (!settings->DynamicChannelArray)
 		goto out_fail;
 
+	settings->TcpKeepAlive = TRUE;
+	settings->TcpKeepAliveRetries = 3;
+	settings->TcpKeepAliveDelay = 2;
```
Comment (mfleisz) [inline libfreerdp/core/settings.c]: Delay used to have value 5 in original code and Interval value 2 - think these two got mixed up in the PR

**Pull request merged**